uroot/root.go

```diff
@@ -26,30 +26,96 @@ const (
 	CmdsPath = "github.com/u-root/u-root/cmds"
 )
 
-// TODO: make this a map so it's easier to find dups.
-type dir struct {
-	name string
-	mode os.FileMode
+type Creator interface {
+	Create() error
+	fmt.Stringer
 }
 
-type file struct {
-	contents string
-	mode     os.FileMode
+type Dir struct {
+	Name string
+	Mode os.FileMode
 }
 
-// TODO: make this a map so it's easier to find dups.
-type dev struct {
-	name    string
-	mode    uint32
-	dev     int
-	howmany int
+func (d Dir) Create() error {
+	return os.MkdirAll(d.Name, d.Mode)
 }
-type mount struct {
-	source string
-	target string
-	fstype string
-	flags  uintptr
-	opts   string
+
+func (d Dir) String() string {
+	return fmt.Sprintf("dir :%q: mode %o", d.Name, d.Mode)
+}
+
+type File struct {
+	Name     string
+	Contents string
+	Mode     os.FileMode
+}
+
+func (f File) Create() error {
+	return ioutil.WriteFile(f.Name, []byte(f.Contents), f.Mode)
+}
+
+func (f File) String() string {
+	return fmt.Sprintf("file %q", f.Name)
+}
+
+type Symlink struct {
+	Target   string
+	Linkpath string
+}
+
+func (s Symlink) Create() error {
+	os.Remove(s.Target)
+	return os.Symlink(s.Linkpath, s.Target)
+}
+
+func (s Symlink) String() string {
+	return fmt.Sprintf("symlink %q -> %q", s.Target, s.Linkpath)
+}
+
+type Link struct {
+	Oldpath string
+	Newpath string
+}
+
+func (s Link) Create() error {
+	os.Remove(s.Newpath)
+	return os.Link(s.Oldpath, s.Newpath)
+}
+
+func (s Link) String() string {
+	return fmt.Sprintf("link %q -> %q", s.Oldpath, s.Newpath)
+}
+
+type Dev struct {
+	Name    string
+	Mode    uint32
+	Dev     int
+	Howmany int
+}
+
+func (d Dev) Create() error {
+	os.Remove(d.Name)
+	return syscall.Mknod(d.Name, d.Mode, d.Dev)
+}
+
+func (d Dev) String() string {
+	return fmt.Sprintf("dev :%q: mode: %#o: magic: %v", d.Name, d.Mode, d.Dev)
+}
+
+type Mount struct {
```

Comment:
I don't see these used outside of this file, you didn't have to export them. But up to you